src/common/cache/__tests__/test.cache.module.ts

```diff
@@ -0,0 +1,10 @@
+import { Global, Module } from '@nestjs/common';
+import { CacheService } from '../cache.service.interface';
+import { FakeCacheService } from './fake.cache.service';
+
+@Global()
+@Module({
+  providers: [{ provide: CacheService, useClass: FakeCacheService }],
```

Comment:
@hectorgomezv similar to the `TestConfigurationModule` I think we should expose a test instance via `useValue`.

I've used `useClass` which works but if we want to manipulate the cache state in a test a reference to it should be provided instead. Wdyt?

Reply:
I'm not sure I get the point, why not use `useValue` then as we do with `TestConfigurationModule` and `TestNetworkModule`? Do you see any cons/impediments?

Reply:
Yes I was considering using the same setup as `TestConfigurationModule`. With `useClass` I think you cannot get to the instance which is being used in this module from the tests. That makes it more difficult to setup a specific cache state.

The "negative" part of this approach is just that you are exposing a global reference to the testing instance. It shouldn't be an issue with our current setup.

I will adjust it to be in the same format as we have `TestConfigurationModule`.

Reply:
Address this here: [5e993fd](https://github.com/5afe/safe-client-gateway-nest/pull/58/commits/5e993fd94c8658e4c9c8a878ca1ad723ac1c22d4)

Just like we use `jest.clearAllMocks()` to reset any mocked state between the tests we now should use `fakeCacheService.clear()` to "reset" the state of the cache between tests.